### ISSUE TITLE
New Rule: implement `markdownlint`'s `MD055`

### ISSUE THREAD

**Issue by tooth-is-silver** [eslint-markdown/eslint-markdown#751]
### Rule details

Add a rule equivalent to markdownlint's MD055 for fGitHub Flavored Markdown tables.

The rule should enforce a consistent use of leading and trailing pipe characters in every table row.

Support the following styles:

- Leading and trailing pipes
- Leading pipes only
- Trailing pipes only
- No leading or trailing pipes
- Consistent

Use `consistent` as the default. It should infer the expected style from the header row of the first table and apply it to every table in the document.

### What type of rule is this?

Enforces a consistent style or user preference

### Example code

```markdown
<!-- Incorrect -->

| Header | Header |
| ------ | ------
  Cell   | Cell   |

<!-- Correct -->

| Header | Header |
| ------ | ------ |
| Cell   | Cell   |
```

### Prior Art

- markdownlint: [`MD055` - Table pipe style](https://github.com/DavidAnson/markdownlint/blob/main/doc/md055.md)
